src/main/java/org/opentripplanner/routing/core/StateEditor.java

```diff
@@ -381,21 +250,13 @@ public void setStartTimeSeconds(long seconds) {
     }
 
     /**
-     * Set non-incremental state values (ex. {@link State#getRoute()}) from an existing state.
-     * Incremental values (ex. {@link State#getNumBoardings()}) are not currently set.
+     * Set non-incremental state values from an existing state.
+     * Incremental values are not currently set.
      * 
      * @param state
      */
     public void setFromState(State state) {
         cloneStateDataAsNeeded();
-        child.stateData.route = state.stateData.route;
-        child.stateData.tripTimes = state.stateData.tripTimes;
-        child.stateData.tripId = state.stateData.tripId;
-        child.stateData.serviceDay = state.stateData.serviceDay;
-        child.stateData.previousTrip = state.stateData.previousTrip;
-        child.stateData.previousStop = state.stateData.previousStop;
-        child.stateData.zone = state.stateData.zone;
-        child.stateData.extensions = state.stateData.extensions;
```

Comment:
Since we're removing so many fields, we may want to just get rid of the StateData and clone the entire state at every iteration of the Astar search. The StateData was introduced when we started getting tons of transit-related fields in State that didn't change often, but it adds complexity.

Reply:
That sounds like a good idea. I didn't want to just go ahead and do it, in case there was some other reason for it existing.

Reply:
Covered in new issue #2879